src/protocol/context/mod.rs

```diff
@@ -67,3 +67,194 @@ pub trait Context<F: Field>:
     /// Generates a new share of one
     fn share_of_one(&self) -> <Self as Context<F>>::Share;
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::ff::{Field, Fp31};
+    use crate::helpers::Direction;
+    use crate::protocol::malicious::Step::MaliciousProtocol;
+    use crate::protocol::{QueryId, RecordId};
+    use crate::secret_sharing::{MaliciousReplicated, Replicated};
+    use crate::telemetry::metrics::{
+        INDEXED_PRSS_GENERATED, RECORDS_SENT, SEQUENTIAL_PRSS_GENERATED,
+    };
+    use futures_util::future::join_all;
+    use futures_util::try_join;
+    use rand::distributions::{Distribution, Standard};
+    use rand::Rng;
+    use std::iter::repeat;
+
+    use super::*;
+    use crate::test_fixture::{Reconstruct, Runner, TestWorld, TestWorldConfig};
+
+    trait AsReplicated<F: Field> {
+        fn left(&self) -> F;
+        fn right(&self) -> F;
+    }
+
+    impl<F: Field> AsReplicated<F> for Replicated<F> {
+        fn left(&self) -> F {
+            (self as &Replicated<F>).left()
+        }
+
+        fn right(&self) -> F {
+            (self as &Replicated<F>).right()
+        }
+    }
+
+    impl<F: Field> AsReplicated<F> for MaliciousReplicated<F> {
+        fn left(&self) -> F {
+            (self as &MaliciousReplicated<F>).rx().left()
```

Comment:
rx() ?

Reply:
I am not `ThisCodeIsAuthorizedToDowngradeFromMalicious` - really the only thing I wanted to do is to send some data using malicious protocol. `rx` works just fine for that and I don't need to pretend I know what I am doing. Let me add a comment explaining  that